File: app/src/main/java/com/mbestavros/geometricweather/daily/adapter/holder/AirQualityHolder.java
package com.mbestavros.geometricweather.daily.adapter.holder;

import android.annotation.SuppressLint;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.core.graphics.ColorUtils;

import com.mbestavros.geometricweather.R;
import com.mbestavros.geometricweather.basic.model.weather.AirQuality;
import com.mbestavros.geometricweather.daily.adapter.DailyWeatherAdapter;
import com.mbestavros.geometricweather.daily.adapter.model.DailyAirQuality;
import com.mbestavros.geometricweather.ui.widget.RoundProgress;

public class AirQualityHolder extends DailyWeatherAdapter.ViewHolder {

    private RoundProgress progress;
    private TextView content;

    public AirQualityHolder(ViewGroup parent) {
        super(LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_weather_daily_air, parent, false));
        progress = itemView.findViewById(R.id.item_weather_daily_air_progress);
        content = itemView.findViewById(R.id.item_weather_daily_air_content);
    }

    @SuppressLint("SetTextI18n")
    @Override
    public void onBindView(DailyWeatherAdapter.ViewModel model, int position) {
        AirQuality airQuality = ((DailyAirQuality) model).getAirQuality();

        int aqi = airQuality.getAqiIndex();
        int color = airQuality.getAqiColor(itemView.getContext());

        progress.setMax(400);
        progress.setProgress(aqi);
        progress.setProgressColor(color);
        progress.setProgressBackgroundColor(
                ColorUtils.setAlphaComponent(color, (int) (255 * 0.1))
        );

        content.setText(aqi + " / " + airQuality.getAqiText());
    }
}

File: app/src/main/java/org/breezyweather/ui/daily/adapter/holder/AirQualityHolder.kt
/**
 * This file is part of Breezy Weather.
 *
 * Breezy Weather is free software: you can redistribute it and/or modify it
 * under the terms of the GNU Lesser General Public License as published by the
 * Free Software Foundation, version 3 of the License.
 *
 * Breezy Weather is distributed in the hope that it will be useful, but
 * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
 * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public
 * License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with Breezy Weather. If not, see <https://www.gnu.org/licenses/>.
 */

package org.breezyweather.ui.daily.adapter.holder

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.TextView
import androidx.core.graphics.ColorUtils
import org.breezyweather.R
import org.breezyweather.domain.weather.model.getColor
import org.breezyweather.domain.weather.model.getIndex
import org.breezyweather.domain.weather.model.getName
import org.breezyweather.ui.common.widgets.RoundProgress
import org.breezyweather.ui.daily.adapter.DailyWeatherAdapter
import org.breezyweather.ui.daily.adapter.model.DailyAirQuality

class AirQualityHolder(parent: ViewGroup) : DailyWeatherAdapter.ViewHolder(
    LayoutInflater.from(parent.context).inflate(R.layout.item_weather_daily_air, parent, false)
) {
    private val mProgress: RoundProgress = itemView.findViewById(R.id.item_weather_daily_air_progress)
    private val mContent: TextView = itemView.findViewById(R.id.item_weather_daily_air_content)

    @SuppressLint("SetTextI18n")
    override fun onBindView(model: DailyWeatherAdapter.ViewModel, position: Int) {
        val airQuality = (model as DailyAirQuality).airQuality
        val aqi = airQuality.getIndex()
        val color = airQuality.getColor(itemView.context)
        mProgress.apply {
            max = 400f
            if (aqi != null) {
                progress = aqi.toFloat()
            }
            setProgressColor(color)
            setProgressBackgroundColor(ColorUtils.setAlphaComponent(color, (255 * 0.1).toInt()))
        }
        mContent.text = aqi.toString() + " / " + airQuality.getName(itemView.context)
    }
}
